src/nifake/unit_tests/test_session.py

```diff
@@ -818,7 +822,12 @@ def setup_method(self, method):
         self.tclk_patched_library_singleton_get = patch('nitclk._library_interpreter._library_singleton.get', return_value=None)
         self.tclk_patched_library_singleton_get.start()
 
-        self.patched_grpc_interpreter.init_with_options.side_effect = [GRPC_SESSION_OBJECT_FOR_TEST]
+        def interpreter_init(*args, **kwargs):
+            self.patched_grpc_interpreter._close_on_exit = True
+            return GRPC_SESSION_OBJECT_FOR_TEST
+
+        self.patched_grpc_interpreter.init_with_options.side_effect = interpreter_init
```

Comment:
I'm guessing we'll never be using this for attaching to an existing session?

Reply:
For the tests, yeah, we set True unconditionally. That may be a testing gap, I'm not sure. This PR is just focused on getting the tests _passing_, though.